test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterHandle.java

```diff
@@ -200,23 +215,7 @@ private void writeUnicastHostsFile() {
             } catch (IOException e) {
                 throw new UncheckedIOException("Failed to write unicast_hosts for: " + node, e);
             }
-        });
-    }
-
-    private void deletePortFiles() {
```

Comment:
Why is deleting ports files specific to full cluster restart tests? wouldn't it also apply in the future to eg rolling restart tests?

Reply:
There's nothing specific to full cluster restart here. This will indeed be used by rolling restart. In fact, we call this method anytime we stop a node for any reason.

Reply:
I've refactored this so `deletePortsFiles()` is called in `Node#stop()` so it's more clear.